daemon/MenuDaemon.vala

```diff
@@ -258,5 +262,78 @@ namespace Gala {
                 push_in = true;
             }, 3, Gdk.CURRENT_TIME);
         }
+
+        public void show_desktop_menu (int x, int y) throws DBusError, IOError {
+            if (desktop_menu == null) {
+                var change_wallpaper = new Gtk.MenuItem.with_label (_("Change Wallpaper…"));
+                change_wallpaper.activate.connect (() => {
+                    try {
+                        AppInfo.launch_default_for_uri ("settings://desktop/appearance/wallpaper", null);
+                    } catch (Error e) {
+                        warning ("Failed to open wallpaper settings: %s", e.message);
+                        var message_dialog  = new Granite.MessageDialog.with_image_from_icon_name (
+                            "Failed to Open Wallpaper Settings",
+                            "The error was: Could not launch Switchboard.",
```

Comment:
We don't ever use "Switchboard" in the UI, so we probably don't want to reference that here. We also want to try to provide some kind of further information about how to resolve the issue or at least begin investigating it. How about
```suggestion
                            "Unable to open System Settings. A handler for the `settings://` URI scheme must be installed.",
```